### PR TITLE
Physics documentation

### PR DESCRIPTION
**Pull request by octotentakel**

### CODE REVIEW

File: Soccar [Physics]/src/nl/soccar/physics/GameEngine.java [hunk 1]
```diff
@@ -132,6 +141,12 @@ private void checkScored() {
         }
     }
 
+    /**
+     * Adds a car to the world.
+     *
+     * @param player
```
Comment (Laugslander) [inline Soccar [Physics]/src/nl/soccar/physics/GameEngine.java]: Parameters documenteren.

File: Soccar [Physics]/src/nl/soccar/physics/GameEngine.java [hunk 2]
```diff
@@ -140,6 +155,11 @@ public void addCar(Player player, CarPhysics car) {
         }
     }
 
+    /**
+     * Removes a car from the world.
+     *
+     * @param player
```
Comment (Laugslander) [inline Soccar [Physics]/src/nl/soccar/physics/GameEngine.java]: Parameter documenteren.

File: Soccar [Physics]/src/nl/soccar/physics/GameEngine.java [hunk 3]
```diff
@@ -148,6 +168,11 @@ public void removeCar(Player player) {
         }
     }
 
+    /**
+     * Adds an object to the world.
+     *
+     * @param object
```
Comment (Laugslander) [inline Soccar [Physics]/src/nl/soccar/physics/GameEngine.java]: Parameter documenteren.

File: Soccar [Physics]/src/nl/soccar/physics/GameEngine.java [hunk 4]
```diff
@@ -162,6 +187,11 @@ public void addWorldObject(WorldObject object) {
         }
     }
 
+    /**
+     * Removes an object from the world.
+     *
+     * @param object
```
Comment (Laugslander) [inline Soccar [Physics]/src/nl/soccar/physics/GameEngine.java]: Parameter documenteren.

File: Soccar [Physics]/src/nl/soccar/physics/models/BallPhysics.java [hunk 5]
```diff
@@ -58,16 +56,35 @@ public BallPhysics(Ball ball, World world) {
...
 
+    /**
+     * Returns the angular velocity
+     *
+     * @return AngularVelocity
+     */
+    public float getAngularVelocity() {
+        return body.getAngularVelocity();
+    }
+
```
Comment (Laugslander) [inline Soccar [Physics]/src/nl/soccar/physics/models/BallPhysics.java]: Enter verwijderen.

File: Soccar [Physics]/src/nl/soccar/physics/models/WheelPhysics.java [hunk 6]
```diff
@@ -177,18 +177,61 @@ public void setAngle(float angle) {
...
+     * @return boolean
+     */
+    public boolean isSteerable() {
+        return steerable;
+    }
+
+    /**
+     * Returns whether this wheel is powered.
+     *
+     * @return boolean
```
Comment (Laugslander) [inline Soccar [Physics]/src/nl/soccar/physics/models/WheelPhysics.java]: Parameter documenteren.

File: Soccar [Physics]/src/nl/soccar/physics/models/WheelPhysics.java [hunk 7]
```diff
@@ -177,18 +177,61 @@ public void setAngle(float angle) {
...
+     * @return The wheel height.
+     */
+    public float getHeight() {
+        return height;
+    }
+
+    /**
+     * Returns whether this wheel is steerable.
+     *
+     * @return boolean
```
Comment (Laugslander) [inline Soccar [Physics]/src/nl/soccar/physics/models/WheelPhysics.java]: Parameter documenteren.

File: Soccar [Physics]/src/nl/soccar/physics/models/ObstaclePhysics.java [hunk 8]
```diff
@@ -52,6 +52,14 @@ public ObstaclePhysics(Obstacle obstacle, World world) {
         body.createFixture(fd);
     }
 
+    /**
+     * Returns the obstacle object.
+     * @return The obstacle object.
```
Comment (Laugslander) [inline Soccar [Physics]/src/nl/soccar/physics/models/ObstaclePhysics.java]: Enter toevoegen tussen beschrijving methode en parameters.

File: Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java [hunk 9]
```diff
@@ -118,6 +117,103 @@ public void setPosition(float x, float y, float degree, float linearVelocityX, f
...
+     * @return The boost trail list
+     */
+    public List<Point2D> getTrail() {
+        return trail;
+    }
+
+    /**
+     * Returns whether the boost is currently active.
+     *
+     * @return boolean
```
Comment (Laugslander) [inline Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java]: Returntype documenteren.

**Review by Laugslander — CHANGES_REQUESTED** [review]
Aantal parameters en returntypen nog documenteren.

**Comment by TehMessiah** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [GameEngine.java#L24](https://github.com/PTS3-S34A/Physics/blob/30457c47c54e1bc379b1197253d0b45899367c74/Soccar [Physics]/src/nl/soccar/physics/GameEngine.java#L24): Remove this unused "room" private field. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1068)
1. MINOR [GameEngine.java#L130](https://github.com/PTS3-S34A/Physics/blob/30457c47c54e1bc379b1197253d0b45899367c74/Soccar [Physics]/src/nl/soccar/physics/GameEngine.java#L130): Replace this "switch" statement by "if" statements to increase readability. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1301)

File: Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java [hunk 10]
```diff
@@ -186,38 +191,77 @@ public float getDegree() {
...
+     *
+     * @return The linear velocity X value.
+     */
     public float getLinearVelocityX() {
         return body.getLinearVelocity().x;
     }
 
+    /**
+     * Returns the linear velocity Y value.
+     *1
```
Comment (Laugslander) [inline Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java]: Random 1 verwijderen.

File: Soccar [Physics]/src/nl/soccar/physics/models/BallPhysics.java [hunk 11]
```diff
@@ -100,16 +101,30 @@ public float getDegree() {
...
+     *
+     * @return The linear velocity X value.
+     */
     public float getLinearVelocityX() {
         return body.getLinearVelocity().x;
     }
 
+    /**
+     * Returns the linear velocity Y value.
+     *1
```
Comment (Laugslander) [inline Soccar [Physics]/src/nl/soccar/physics/models/BallPhysics.java]: Random 1 verwijderen.

**Review by Laugslander — CHANGES_REQUESTED** [review]

**Review by Laugslander — APPROVED** [review]

**Comment by BeeHiveJava** [comment]
retest this please

**Pull request merged**